hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/validate/HazelcastSqlValidator.java

```diff
@@ -563,13 +565,36 @@ public CalciteContextException newValidationError(SqlNode node, Resources.ExInst
             Object[] arguments = ResourceUtil.args(e);
             String identifier = (arguments != null && arguments.length > 0) ? String.valueOf(arguments[0]) : null;
             Mapping mapping = identifier != null ? iMapResolver.resolve(identifier) : null;
-            String sql = mapping != null ? SqlCreateMapping.unparse(mapping) : null;
+            String sql = mapping != null && hasMapAccess(identifier) ? SqlCreateMapping.unparse(mapping) : null;
```

Comment:
the more correct order would be first to check permissions and then invoke `iMapResolver.resolve(identifier)`